Repositories/ProductSqlServerDBRepository.cs

```diff
@@ -0,0 +1,151 @@
+﻿using InventoryManagement.Models;
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.IO;
+
+namespace InventoryManagement.Repositories
+{
+    public class ProductSqlServerDBRepository:IProductRepository,IDisposable
+    {
+        private SqlConnection _connection;
+   
+        public void SetupDB()
+        {
+            _connection = new();
+           try
+            {
+                SetUpConnectionString();
+                _connection.Open();
+            }
+            catch { throw; }
+
+        }
+        private void SetUpConnectionString()
+        {
+            var config = new ConfigurationBuilder().SetBasePath(Directory.GetCurrentDirectory()).AddJsonFile("appsettings.json").Build();
+            string connectionString = config["SqlServerDB:connectionString"];
+            _connection.ConnectionString = connectionString;
+        }
+
+        private void CloseConnection()
+        {
+            if (_connection.State != ConnectionState.Closed)
+                try { _connection.Close(); }
+                catch { throw; }
+
+        }
+        public void Dispose()
+        {
+            CloseConnection();
+            _connection.Dispose();
+        }
+        public IEnumerable<Product> GetProducts()
+        { 
+            string query = $"select * from Inventory";
+            using (SqlCommand cmd = new(query, _connection))
+            {
+                using (SqlDataReader reader = cmd.ExecuteReader())
```

Comment:
you can use the new `using` syntax, which can be done without curly braces.
always try to avoid nested if you can.